worker/changestream/metrics.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package changestream
+
+import "github.com/prometheus/client_golang/prometheus"
+
+const (
+	changestreamMetricsNamespace   = "juju"
+	changestreamSubsystemNamespace = "db"
+)
+
+// Collector defines a prometheus collector for the dbaccessor.
+type Collector struct {
+	// Stream metrics.
+	WatermarkInserts       *prometheus.CounterVec
+	WatermarkRetries       *prometheus.CounterVec
+	ChangesRequestDuration *prometheus.HistogramVec
+	ChangesCount           *prometheus.HistogramVec
+	// EventMultiplexer metrics.
+	Subscriptions  prometheus.Gauge
+	DispatchErrors *prometheus.CounterVec
+}
+
+func (c *Collector) WatermarkInsertsInc() {
+	c.WatermarkInserts.WithLabelValues().Inc()
```

Comment:
We should add label values here. I'd at least want to see the namespace so that we can see which ones are at fault.

Reply:
done